numpy/core/setup.py

```diff
@@ -184,20 +185,12 @@ def check_funcs(funcs_name, headers=["feature_detection_math.h"]):
     # config.h in the public namespace, so we have a clash for the common
     # functions we test. We remove every function tested by python's
     # autoconf, hoping their own test are correct
-    for f in OPTIONAL_STDFUNCS_MAYBE:
-        if config.check_decl(fname2def(f),
-                    headers=["Python.h", "math.h"]):
-            if f in OPTIONAL_STDFUNCS:
-                OPTIONAL_STDFUNCS.remove(f)
-            else:
-                OPTIONAL_FILE_FUNCS.remove(f)
+    for f in OPTIONAL_FUNCS_MAYBE:
+        if config.check_decl(fname2def(f), headers=["Python.h"]):
+            OPTIONAL_FILE_FUNCS.remove(f)
```

Comment:
I had a quick peek at this; it's still the case that CPython does this. There's other defines too, like `HAVE_ASINH`, but those are highly unlikely to contain a different value at numpy build time vs. Python build time. Not sure if this check is still needed, but fine to leave as is.

Reply:
Also: `:wags-finger-at-python:`